### PR TITLE
Drop Scala 2.12 compatibility layer from //daml-lf/data

### PR DESCRIPTION
**Pull request by cocreature**
changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/InsertOrdSet.scala [hunk 1]
```diff
@@ -40,16 +46,35 @@ final class InsertOrdSet[T] private (_items: Queue[T], _hashSet: HashSet[T])
...
-  // Here only for 2.12 (harmless in 2.13); placed in InsertOrdSetCompanion the
-  // implicit gets in an unwinnable fight with Set's version
-  implicit override def canBuildFrom[A]: Factory[A] =
-    super.canBuildFrom
-
+  private final class InsertOrdSetBuilder[T] extends ReusableBuilder[T, InsertOrdSet[T]] {
+    var m: InsertOrdSet[T] = empty;
+    override def clear(): Unit = {
+      m = empty;
+    }
```
Comment (remyhaemmerle-da) [inline daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/InsertOrdSet.scala:73]: Those `;` are a bit strange.

File: daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/ImmArray.scala [hunk 2]
```diff
@@ -11,7 +11,10 @@ import scalaz.syntax.applicative._
 import scalaz.{Applicative, Equal, Foldable, Order, Traverse}
 
 import scala.annotation.tailrec
+import scala.collection.StrictOptimizedSeqFactory
 import scala.collection.compat.immutable.ArraySeq
```
Comment (remyhaemmerle-da) [inline daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/ImmArray.scala:15]:
```suggestion
import scala.collection.immutable.ArraySeq
```

**Review by remyhaemmerle-da — COMMENTED** [review]
Thanks. 

I think we could get rid of `@maven//:org_scala_lang_modules_scala_collection_compat`

File: daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/InsertOrdSet.scala (hunk 1 above)
Comment (cocreature) [reply daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/InsertOrdSet.scala]: just copied the code around not sure why they were there before. Killed them now

**Comment by cocreature** [comment]
> I think we could get rid of `@maven//:org_scala_lang_modules_scala_collection_compat`

dropped all dependencies on the compat lib

**Pull request merged**